sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_async.py

```diff
@@ -171,54 +175,67 @@ def _next_message_in_buffer(self):
+            except asyncio.CancelledError:  # pylint: disable=try-except-raise
+                raise
+            except Exception as exception:  # pylint: disable=broad-except
+                if (
                         isinstance(exception, error.AMQPLinkError)
                         and exception.condition == error.ErrorCondition.LinkStolen  # pylint: disable=no-member
-                    ):
-                        raise await self._handle_exception(exception)
-                    if not self.running:  # exit by close
-                        return
-                    if self._last_received_event:
-                        self._offset = self._last_received_event.offset
-                    last_exception = await self._handle_exception(exception)
-                    retried_times += 1
-                    if retried_times > max_retries:
-                        _LOGGER.info(
-                            "%r operation has exhausted retry. Last exception: %r.",
-                            self._name,
-                            last_exception,
-                        )
-                        raise last_exception
-
-        if self._message_buffer:
-            while self._message_buffer:
-                if batch:
-                    events_for_callback = []  # type: List[EventData]
-                    for _ in range(min(max_batch_size, len(self._message_buffer))):
-                        events_for_callback.append(self._next_message_in_buffer())
-                    await self._on_event_received(events_for_callback)
-                else:
-                    await self._on_event_received(self._next_message_in_buffer())
-        elif max_wait_time:
-            if batch:
-                await self._on_event_received([])
-            else:
-                await self._on_event_received(None)
+                ):
+                    raise await self._handle_exception(exception)
+                if not self.running:  # exit by close
+                    return
+                if self._last_received_event:
+                    self._offset = self._last_received_event.offset
+                last_exception = await self._handle_exception(exception)
+                retried_times += 1
+                if retried_times > max_retries:
+                    _LOGGER.info(
+                        "%r operation has exhausted retry. Last exception: %r.",
+                        self._name,
+                        last_exception,
+                    )
+                    raise last_exception
+
+    async def receive(self, batch=False, max_batch_size=300, max_wait_time=None) -> None:
+        self._callback_task_run = True
+        self._last_callback_called_time = time.time()
+        callback_task = asyncio.ensure_future(self._callback_coroutine(batch, max_batch_size, max_wait_time))
+        receive_task = asyncio.ensure_future(self._receive_coroutine())
```

Comment:
Nitpick: Not sure I'm a fan of using `_coroutine` as a method name suffix....

Reply:
I could just rename it as `_task` :P

Reply:
updated to `_task`